Internal Server Error 500 when using  SSL self-signed S3-Compatible Object Storage

## Problem

When using Habor Helm Chart to deploy Habor to Kubernetes with SSL self-signed S3-Compatible Object Storage configuration, I got 500 Internal Error when access Harbor after deloyed: `x509: cannot validate certificate for xxx because it doesn't contain any IP SANs`

This is values passed to Habor Helm Chart:

```yaml
  imageChartStorage:
    type: s3
    s3:
      region: default
      bucket: "harbor"
      accesskey: "ACCESS_KEY"
      secretkey: "SECRET_KEY"
      regionendpoint: https://s3-ceph.lab1.cluster1:8087
      encrypt: false
      secure: true
      skipverify: true
      v4auth: true
      multipartcopythresholdsize: "5368709120"
    disableredirect: false

```

This is log from harbor core and chart museum containers:

Harbor chart museum container log:

```log
{"L":"ERROR","T":"2022-06-16T07:01:23.808Z","M":"[11] RequestError: send request failed\ncaused by: Get \"https://s3-ceph.lab1.cluster1:8087/harbor?prefix=library\": x509: cannot validate certificate for s3-ceph.lab1.cluster1 because it doesn't contain any IP SANs","repo":"library","reqID":"10572bf4-0c96-45ae-92c7-39e31593926b"}
{"L":"ERROR","T":"2022-06-16T07:01:23.808Z","M":"[11] Request served","path":"/library/index.yaml","comment":"","clientIP":"172.16.22.57","method":"GET","statusCode":500,"latency":"408.532628ms","reqID":"10572bf4-0c96-45ae-92c7-39e31593926b"}
{"L":"ERROR","T":"2022-06-16T07:01:23.859Z","M":"[10] RequestError: send request failed\ncaused by: Get \"https://s3-ceph.lab1.cluster1:8087/harbor?prefix=project01\": x509: cannot validate certificate for s3-ceph.lab1.cluster1 because it doesn't contain any IP SANs","repo":"project01","reqID":"02ac4129-1e55-4cea-a48c-3ade065e4a53"}
{"L":"ERROR","T":"2022-06-16T07:01:23.859Z","M":"[10] Request served","path":"/project01/index.yaml","comment":"","clientIP":"172.16.22.57","method":"GET","statusCode":500,"latency":"460.001389ms","reqID":"02ac4129-1e55-4cea-a48c-3ade065e4a53"}
{"L":"ERROR","T":"2022-06-16T07:01:26.818Z","M":"[13] RequestError: send request failed\ncaused by: Get \"https://s3-ceph.lab1.cluster1:8087/harbor?prefix=project01\": x509: cannot validate certificate for s3-ceph.lab1.cluster1 because it doesn't contain any IP SANs","repo":"project01","reqID":"5e533557-13ee-488e-8163-db10ad2768fc"}
{"L":"ERROR","T":"2022-06-16T07:01:26.818Z","M":"[13] Request served","path":"/project01/index.yaml","comment":"","clientIP":"172.16.22.57","method":"GET","statusCode":500,"latency":"353.319276ms","reqID":"5e533557-13ee-488e-8163-db10ad2768fc"}
{"L":"ERROR","T":"2022-06-16T07:01:32.820Z","M":"[15] RequestError: send request failed\ncaused by: Get \"https://s3-ceph.lab1.cluster1:8087/harbor?prefix=library\": x509: cannot validate certificate for s3-ceph.lab1.cluster1 because it doesn't contain any IP SANs","repo":"library","reqID":"a0e6c505-d80c-4c68-ac24-c62344656638"}
{"L":"ERROR","T":"2022-06-16T07:01:32.821Z","M":"[15] Request served","path":"/library/index.yaml","comment":"","clientIP":"172.16.22.57","method":"GET","statusCode":500,"latency":"332.671572ms","reqID":"a0e6c505-d80c-4c68-ac24-c62344656638"}
{"L":"ERROR","T":"2022-06-16T07:01:32.825Z","M":"[16] RequestError: send request failed\ncaused by: Get \"https://s3-ceph.lab1.cluster1:8087/harbor?prefix=project01\": x509: cannot validate certificate for s3-ceph.lab1.cluster1 because it doesn't contain any IP SANs","repo":"project01","reqID":"db3b379b-df6a-498f-929e-835ca65bdda7"}
{"L":"ERROR","T":"2022-06-16T07:01:32.825Z","M":"[16] Request served","path":"/project01/index.yaml","comment":"","clientIP":"172.16.22.57","method":"GET","statusCode":500,"latency":"337.075279ms","reqID":"db3b379b-df6a-498f-929e-835ca65bdda7"}
{"L":"ERROR","T":"2022-06-16T07:01:36.261Z","M":"[17] RequestError: send request failed\ncaused by: Get \"https://s3-ceph.lab1.cluster1:8087/harbor?prefix=project01\": x509: cannot validate certificate for s3-ceph.lab1.cluster1 because it doesn't contain any IP SANs","repo":"project01","reqID":"6853a6bc-2c11-4ae1-9453-cfdc633888f2"}
{"L":"ERROR","T":"2022-06-16T07:01:36.261Z","M":"[17] Request served","path":"/project01/index.yaml","comment":"","clientIP":"172.16.22.57","method":"GET","statusCode":500,"latency":"398.517184ms","reqID":"6853a6bc-2c11-4ae1-9453-cfdc633888f2"}```
```

Harbor core container log:

```log
caused by: Get "https://s3-ceph.lab1.cluster1:8087/harbor?prefix=library": x509: cannot validate certificate for s3-ceph.lab1.cluster1 because it doesn't contain any IP SANs
2022-06-16T07:01:23Z [ERROR] [/server/v2.0/handler/project.go:505][requestID="0bced62c-6e13-4a9b-b806-6d1363194ac5"]: failed to populate properties for project project01, error: get chart count of project 2 failed: http error: code 500, message RequestError: send request failed
caused by: Get "https://s3-ceph.lab1.cluster1:8087/harbor?prefix=project01": x509: cannot validate certificate for s3-ceph.lab1.cluster1 because it doesn't contain any IP SANs
2022-06-16T07:01:26Z [ERROR] [/lib/http/error.go:54]: {"errors":[{"code":"UNKNOWN","message":"get chart count of project 2 failed: http error: code 500, message RequestError: send request failed\ncaused by: Get \"https://s3-ceph.lab1.cluster1:8087/harbor?prefix=project01\": x509: cannot validate certificate for s3-ceph.lab1.cluster1 because it doesn't contain any IP SANs"}]}
/harbor/src/server/v2.0/handler/project.go:727, github.com/goharbor/harbor/src/server/v2.0/handler.(*projectAPI).populateProperties
/harbor/src/server/v2.0/handler/project.go:666, github.com/goharbor/harbor/src/server/v2.0/handler.(*projectAPI).getProject
/harbor/src/server/v2.0/handler/project.go:335, github.com/goharbor/harbor/src/server/v2.0/handler.(*projectAPI).GetProject
/harbor/src/server/v2.0/restapi/configure_harbor.go:2135, github.com/goharbor/harbor/src/server/v2.0/restapi.HandlerAPI.func92
/harbor/src/server/v2.0/restapi/operations/project/get_project.go:19, github.com/goharbor/harbor/src/server/v2.0/restapi/operations/project.GetProjectHandlerFunc.Handle
/harbor/src/server/v2.0/restapi/operations/project/get_project.go:69, github.com/goharbor/harbor/src/server/v2.0/restapi/operations/project.(*GetProject).ServeHTTP
/harbor/src/server/v2.0/restapi/operations/harbor_api.go:1762, github.com/goharbor/harbor/src/server/v2.0/restapi/operations.(*HarborAPI).handler.func1
/harbor/src/server/middleware/apiversion/api_version.go:28, github.com/goharbor/harbor/src/server/middleware/apiversion.Middleware.func1.1
/usr/local/go/src/net/http/server.go:2047, net/http.HandlerFunc.ServeHTTP
```

The reason is harbor registry container is configured for  SSL self-signed S3-Compatible Object Storage, but chart museum container is not configured correctly to support SSL self-signed S3-Compatible Object Storage.

The solution is add following env variable:

```yaml
kind: ConfigMap
apiVersion: v1
metadata:
  name: harbor-chartmuseum
data:
  # ...
  AWS_INSECURE_SKIP_VERIFY: "true"
```

to chart museum ConfigMap when `skipverify` is set to `true` in Helm chart

## Refferences

- https://github.com/helm/chartmuseum/issues/162#issuecomment-1137528877

--- a/templates/chartmuseum/chartmuseum-cm.yaml
+++ b/templates/chartmuseum/chartmuseum-cm.yaml
@@ -69,6 +69,9 @@ data:
   {{- if $storage.s3.regionendpoint }}
   STORAGE_AMAZON_ENDPOINT: {{ $storage.s3.regionendpoint }}
   {{- end }}
+  {{- if $storage.s3.skipverify }}
+  AWS_INSECURE_SKIP_VERIFY: "true"
+  {{- end }}
   {{- if and (not $storage.s3.existingSecret) ($storage.s3.accesskey) }}
   AWS_ACCESS_KEY_ID: {{ $storage.s3.accesskey }}
   {{- end }}

